File: 2553-separate-the-digits-in-an-array/2553-separate-the-digits-in-an-array.cpp
class Solution {
public:
    vector<int> separateDigits(vector<int>& nums) {
        vector<int> result;
        stack<int> st;
        
        for(int i=0; i<nums.size(); i++)
        {
            int temp=nums[i];
            while(temp!=0)
            {
                int digit=temp%10;
                st.push(digit);
                temp/=10;
            }
            while(!st.empty())
            {
                result.push_back(st.top());
                st.pop();
            }
        }
        return result;
    }
};

File: 2553-separate-the-digits-in-an-array/2553-separate-the-digits-in-an-array.java
class Solution {
    public int[] separateDigits(int[] nums) {
        String s="";
        for(int i=0; i<nums.length;i++){
            s+=String.valueOf(nums[i]);
            }
        int arr[]=new int [s.length()];
        for(int i=0; i<s.length();i++)
        {
            arr[i]=s.charAt(i)-'0';
        }
        return arr;
    }
}